server/index.js

```diff
@@ -31,4 +31,27 @@ app.post('/reviews', (req, res) => {
   });
 });
 
+app.put('/reviews', (req, res) => {
+  let query = {_id: req.body._id};
+  let vote = req.body.vote;
+
+  Reviews.findOne(query)
+    .then((review) => Number(review.popularity))
+    .then((popularity) => {
+      if (vote === 'upvote') {
+        popularity++;
+        return popularity;
+      } else if (vote === 'downvote') {
+        popularity--;
+        return popularity--;
```

Comment:
It looks like you're decrementing `popularity` on both line 45 and and line 46, which I don't think is intended. Also, you could have just one return statement after the conditional.